Mathlib/CategoryTheory/Category/Quiv.lean

```diff
@@ -82,6 +82,38 @@ end Cat
 
 namespace Quiv
 
+def isoOfEquiv {V W : Type u } [Quiver.{v + 1, u} V] [Quiver.{v + 1, u} W]
```

Comment:
I believe it would much more convenient to develop a suitable API (kind of similar to `eqToHom`) for quivers:
```lean
section

variable {V : Type*} [Quiver V]

def _root_.Quiver.homOfEq {X Y : V} (f : X ⟶ Y) {X' Y' : V}
    (hX : X = X') (hY : Y = Y') : X' ⟶ Y' := by
  subst hX hY
  exact f

@[simp]
lemma _root_.Quiver.homOfEq_trans
    {X Y : V} (f : X ⟶ Y) {X' Y' : V} (hX : X = X') (hY : Y = Y')
    {X'' Y'' : V} (hX' : X' = X'') (hY' : Y' = Y'') :
    Quiver.homOfEq (Quiver.homOfEq f hX hY) hX' hY' =
      Quiver.homOfEq f (hX.trans hX') (hY.trans hY') := by
  subst hX hY hX' hY'
  rfl

lemma _root_.Quiver.homOfEq_injective {X X' Y Y' : V} (hX : X = X') (hY : Y = Y')
    {f g : X ⟶ Y} (h : Quiver.homOfEq f hX hY = Quiver.homOfEq g hX hY) : f = g := by
  subst hX hY
  exact h

@[simp]
lemma _root_.Quiver.homOfEq_rfl {X Y : V} (f : X ⟶ Y) :
    Quiver.homOfEq f rfl rfl = f := rfl

end

-- this lemma may replace `Prefunctor.ext`
theorem _root_.Prefunctor.ext' {V W : Type u} [Quiver V] [Quiver W] {F G : Prefunctor V W}
    (h_obj : ∀ X, F.obj X = G.obj X)
    (h_map : ∀ (X Y : V) (f : X ⟶ Y),
      F.map f = Quiver.homOfEq (G.map f) (h_obj _).symm (h_obj _).symm) : F = G := by
  obtain ⟨Fobj, Fmap⟩ := F
  obtain ⟨Gobj, Gmap⟩ := G
  obtain rfl : Fobj = Gobj := funext h_obj
  simp only [Prefunctor.mk.injEq, heq_eq_eq, true_and]
  ext X Y f
  simpa only [Quiver.homOfEq_rfl] using h_map X Y f
```
Then, `isoOfEquiv` becomes:
```lean
namespace Quiv

section
variable {V W : Type u } [Quiver V] [Quiver W]
  (e : V ≃ W) (he : ∀ X Y : V, (X ⟶ Y) ≃ (e X ⟶ e Y))

include he in
@[simp]
lemma homOfEq_map_homOfEq {X Y : V} (f : X ⟶ Y) {X' Y' : V} (hX : X = X') (hY : Y = Y')
    {X'' Y'' : W} (hX' : e X' = X'') (hY' : e Y' = Y'') :
    Quiver.homOfEq (he _ _ (Quiver.homOfEq f hX hY)) hX' hY' =
      Quiver.homOfEq (he _ _ f) (by rw [hX, hX']) (by rw [hY, hY']) := by
  subst hX hY hX' hY'
  rfl

def isoOfEquiv : Quiv.of V ≅ Quiv.of W where
      hom := Prefunctor.mk e (he _ _)
      inv :=
        { obj := e.symm
          map {X Y} f := (he _ _).symm (Quiver.homOfEq f (by simp) (by simp)) }
      hom_inv_id := Prefunctor.ext' e.left_inv (fun X Y f ↦ by
        dsimp [Quiv.id_eq_id, Quiv.comp_eq_comp]
        apply (he _ _).injective
        apply Quiver.homOfEq_injective (X' := e X) (Y' := e Y) (by simp) (by simp)
        simp)
      inv_hom_id := Prefunctor.ext' e.right_inv (by simp [Quiv.id_eq_id, Quiv.comp_eq_comp])

end
```